wasm/jsapi/table/constructor-reftypes.tentative.any.js

```diff
@@ -0,0 +1,49 @@
+// META: global=window,dedicatedworker,jsshell
+// META: script=assertions.js
+// META: script=/wasm/jsapi/wasm-constants.js
+// META: script=/wasm/jsapi/wasm-module-builder.js
+
+// Test cases for changes to the WebAssembly.Table constructor API that
+// come in with the reftypes proposal: the API takes a default argument, which
+// is used as an initializing value for the WebAssembly.Table.
+//
+// See:
+//   https://webassembly.github.io/reference-types/js-api/index.html#tables
+
+test(() => {
+  const testObject = {};
+  const argument = { "element": "externref", "initial": 3 };
+  const table = new WebAssembly.Table(argument, testObject);
+  assert_equals(table.length, 3);
+  assert_equals(table.get(0), testObject);
+  assert_equals(table.get(1), testObject);
+  assert_equals(table.get(2), testObject);
+}, "initialize externref table with default value");
+
+test(() => {
+  const argument = { "element": "externref", "initial": 3 };
+  assert_throws_js(TypeError, () => new WebAssembly.Table(argument, 37));
+}, "initialize externref table with i32 default value");
```

Comment:
- Why does this throw?
- I meant testing with `{ "element": "i32" }`, more than the value itself

Reply:
ah, sorry for this mess,
It shouldn't throw for externref's tables because boxed value of 37 is acceptable externref